identity-iota/src/tangle/resolver.rs

```diff
@@ -124,11 +136,104 @@ where
   /// # Errors
   ///
   /// Errors if the holder URL is missing, is not a valid [`IotaDID`], or DID resolution fails.
-  pub async fn resolve_presentation_holder(&self, presentation: &Presentation) -> Result<ResolvedIotaDocument> {
-    let holder_url: &Url = presentation.holder.as_ref().ok_or(Error::InvalidPresentationHolder)?;
+  pub async fn resolve_presentation_holder<U, V>(
+    &self,
+    presentation: &Presentation<U, V>,
+  ) -> Result<ResolvedIotaDocument> {
+    let holder_url: &Url = presentation.holder.as_ref().ok_or(Error::IsolatedValidationError(
+      crate::credential::ValidationError::MissingPresentationHolder,
+    ))?;
     let holder: IotaDID = IotaDID::parse(holder_url.as_str())?;
     self.resolve(&holder).await
   }
+
+  /// Verifies a [`Credential`].
+  ///
+  /// This method resolves the issuer's DID Document and validates the following properties in accordance with
+  /// `options`:
+  /// - The issuer's signature
+  /// - The expiration date
+  /// - The issuance date
+  /// - The credential's semantic structure.
+  ///
+  /// If you already have an up to date version of the issuer's resolved DID Document you may want to use
+  /// [`CredentialValidator::validate`](CredentialValidator::validate()) in order to avoid an unnecessary resolution.
+  ///
+  /// # Warning
+  ///  There are many properties defined in [The Verifiable Credentials Data Model](https://www.w3.org/TR/vc-data-model/) that are **not** validated.
+  ///  Examples of properties **not** validated by this method includes: credentialStatus, types, credentialSchema,
+  /// refreshService **and more**.
+  ///
+  /// # Errors
+  /// If the issuer's DID Document cannot be resolved an error will be returned immediately. Otherwise
+  /// an attempt will be made to validate the credential. If the `fail_fast` parameter is "Yes" an error will be
+  /// returned upon the first encountered validation failure, otherwise all validation errors will be accumulated in
+  /// the returned error.
+  pub async fn verify_credential<U: Serialize>(
+    &self,
+    credential: &Credential<U>,
+    options: &CredentialValidationOptions,
+    fail_fast: FailFast,
+  ) -> Result<()> {
+    let issuer = self.resolve_credential_issuer(credential).await?;
+    CredentialValidator::validate(credential, options, &issuer, fail_fast).map_err(Into::into)
+  }
```

Comment:
I have a bit mixed feelings about this. On the one hand it is nice to have consistency with `verify_presentation`, on the other hand I would like to encourage the use of the sync version when async is not needed. I don't feel very strongly about this however so might as well just go with your suggestion as I always really appreciate review comments.

Reply:
I see the argument but I believe we may tend towards `async` validations for the `Credential` in the future anyway, depending on if we choose to validate the schema or revocation `status`. I.e. if we have to resolve the `credentialSchema` or support `RevocationList` it means an external, async, lookup during validation; if we support some sort of `DocumentRevocationBitmap` which only needs the issuer document then it may stay synchronous.

Secondly, wouldn't developers have to perform an async resolution of the issuer in most cases anyway? Performance-conscious users can still use `CredentialValidator` but for the majority we should just use the `async` methods from `Resolver`.